python-package/lightgbm/sklearn.py

```diff
@@ -673,6 +673,18 @@ def _more_tags(self) -> Dict[str, Any]:
             },
         }
 
+    def __sklearn_tags__(self):
+        tags = super().__sklearn_tags__()
```

Comment:
Maybe a comment like this would be useful:

> scikit-learn introduced `__sklearn_tags__` in version 1.6. Remove `_more_tags` when minimum supported scikit-learn version is 1.6.

Reply:
Yes, please add a comment like this on `_more_tags` (with a reference to this PR).

Reply:
Please use exactly this text:

```python
# scikit-learn 1.6 introduced an __sklearn__tags() method intended to replace _more_tags().
# _more_tags() can be removed whenever lightgbm's minimum supported scikit-learn version
# is >=1.6.
# ref: https://github.com/microsoft/LightGBM/pull/6651
```

Reply:
Added

Reply:
Thanks! I'd intended for the comment to be on the definition of `_more_tags()` (apologies for not being clearer about that).

I just pushed https://github.com/microsoft/LightGBM/pull/6651/commits/c4ec9a4939894885d8ec9eee8dd1911c62d40107 moving it.